GoogleForCreators/web-stories-wp
# Quick Actions: Default actions

## Context

Quick action menus

## Summary

Fix the default actions to work as expected:
1. Clicking the 'insert background media' action should focus the media tab button in the library nav bar
2. Clicking the 'insert text' action should focus the text tab button in the library nav bar

## Relevant Technical Choices

The `useFocusHighlight` and `useFocusOut` hooks need refs to work. The refs must look like `{ current: <element> }`. However, the `TabView` component used refs, but the refs were not the same shape. They looked like `{ current: { [id: string]: <element>} }`.

To use the existing hooks I had to change the way the tab view refs were set up. This was a little strange though, because the tabs are dynamically set, so we couldn't create `n` number of refs and have this change.

To get around this, we are now creating a ref for every tab but may not be using all of them (if some are not rendered). I tried to put this all in the same place so that it's easy to update in the future (in `LibraryProvider` and `InspectorProvider`).

## To-do

n/a

## User-facing changes

Before the focus would stay in the quick action menu. Now the library tabs focus when we use keyboard.

|Before|After|
|--|--|
|![click](https://user-images.githubusercontent.com/22185279/118197937-2a2b7e00-b40d-11eb-9c88-c6bd0cdafd2b.gif)|![click](https://user-images.githubusercontent.com/22185279/118197842-fbada300-b40c-11eb-89bf-5b981c8cd19e.gif)|
|![keyboard-navigation](https://user-images.githubusercontent.com/22185279/118197967-3ca5b780-b40d-11eb-8608-5648d1240550.gif)|![keyboard-navigation](https://user-images.githubusercontent.com/22185279/118197834-f7818580-b40c-11eb-8e04-3b1b6644ab2b.gif)|

## Testing Instructions

1. Make sure the quick actions experiment is turned on
2. Go to editor, don't select an element
3. Tab into quick action menu
4. Press `space` or `enter` on the bottom two actions.
5. Should have the focus taken to the correct tab in the library

### QA






### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #7512
